Marlin/src/module/temperature.cpp

```diff
@@ -2246,65 +2316,83 @@ void Temperature::disable_all_heaters() {
-      if (i > 0) max6675_temp <<= 8; // shift left if not the last byte
-    }
+    #if !HAS_MAX31865 && !HAS_MAX31855 && !HAS_MAX6675_TEMP
+      for (uint8_t i = sizeof(max6675_temp); i--;) {
+        max6675_temp |= TERN(MAX6675_SEPARATE_SPI, max6675_spi.receive(), spiRec());
+        if (i > 0) max6675_temp <<= 8; // shift left if not the last byte
+      }
+        MAX6675_WRITE(HIGH); // disable TT_MAX6675
+    #endif
+
+    #if HAS_MAX31855
+      Adafruit_MAX31855 &max855ref = MAX6675_SEL(max31855_0, max31855_1);  
+      max6675_temp = max855ref.readRaw32();  
+    #endif
 
-    MAX6675_WRITE(HIGH); // disable TT_MAX6675
+    #if HAS_MAX31865
+      Adafruit_MAX31865 &max865ref = MAX6675_SEL(max31865_0, max31865_1);
+      max6675_temp = max865ref.readRTD_with_Fault();
+    #endif
+
+    #if HAS_MAX6675_TEMP
+      MAX6675 &max6675ref = MAX6675_SEL(max6675_0, max6675_1);
+      max6675_temp = max6675ref.readRaw16();
+    #endif
+    
+    // At the present time we do not have the ability to set the MAX31865 HIGH threshold 
+    // or thr LOW threshold, so no need to check for them, zero these bits out
+    const uint8_t fault_31865 = TERN1(HAS_MAX31865, (max865ref.readFault() & 0x3FU));
 
-    const uint8_t fault_31865 = TERN1(HAS_MAX31865, maxref.readFault());
 
     if (DISABLED(IGNORE_THERMOCOUPLE_ERRORS) && (max6675_temp & MAX6675_ERROR_MASK) && fault_31865) {
       max6675_errors[hindex]++;
       if (max6675_errors[hindex] > THERMOCOUPLE_MAX_ERRORS) {
         SERIAL_ERROR_START();
         SERIAL_ECHOPGM("Temp measurement error! ");
         #if MAX6675_ERROR_MASK == 7
-          SERIAL_ECHOPGM("MAX31855 ");
-          if (max6675_temp & 1)
-            SERIAL_ECHOLNPGM("Open Circuit");
-          else if (max6675_temp & 2)
-            SERIAL_ECHOLNPGM("Short to GND");
-          else if (max6675_temp & 4)
-            SERIAL_ECHOLNPGM("Short to VCC");
-        #elif HAS_MAX31865
+            SERIAL_ECHOPGM("MAX31855 ");
+            if (max6675_temp & 1)
+              SERIAL_ECHOLNPAIR("Fault :(", max6675_temp & 1 ,")  >> Open Circuit");
+            else if (max6675_temp & 2)
+              SERIAL_ECHOLNPAIR("Fault : (", max6675_temp & 2 ,")  >> Short to GND");
+            else if (max6675_temp & 4)
+              SERIAL_ECHOLNPAIR("Fault : (", max6675_temp & 4 ,")  >> Short to VCC");
+        #elif MAX6675_HAS_MAX31865
+          const uint8_t fault_31865 = max865ref.readFault(); 
+            max865ref.clearFault();        
           if (fault_31865) {
-            maxref.clearFault();
-            SERIAL_ECHOPAIR("MAX31865 Fault :(", fault_31865, ")  >>");
+            SERIAL_ECHOLNPAIR("\nMAX31865 Fault :(", fault_31865, ")  >>");
             if (fault_31865 & MAX31865_FAULT_HIGHTHRESH)
               SERIAL_ECHOLNPGM("RTD High Threshold");
-            else if (fault_31865 & MAX31865_FAULT_LOWTHRESH)
+            if (fault_31865 & MAX31865_FAULT_LOWTHRESH)
```

Comment:
can more than one fault happen at same time?

Reply:
Yes, on the MAX31865 board